### ISSUE TITLE
Can't use patch source(s) with ssm_patch_baseline

### ISSUE THREAD

**Issue by karnauskas** [hashicorp/terraform-provider-aws#4748]
Looks like terraform module is missing configuration parameters to specify alternative patch sources/repositories.

This is available via sdk/api: https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SSM/Types/PatchSource.html.

**Comment by jdheyburn** [comment]
This is feasible - although it is only for Linux OSs - I trust AWS are validating this against their API.

* https://docs.aws.amazon.com/systems-manager/latest/userguide/patch-manager-how-it-works-alt-source-repository.html

Proposed configuration:

```hcl
resource "aws_ssm_patch_baseline" "production" {
  name             = "patch-baseline"
  description      = "Patch Baseline Description"
  operating_system = "CENTOS"

  # New param below
  patch_sources = ["abc123.com"]

  ...
}
```

Will try to pick this up in next few weeks.

**Comment by jdheyburn** [comment]
Revising the proposed configuration for this:

```hcl
resource "aws_ssm_patch_baseline" "production" {
  name             = "patch-baseline"
  description      = "Patch Baseline Description"
  operating_system = "CENTOS"

  # New param below
  patch_source {
    configuration = <<EOF
        <configuration_here>
    EOF

    name = "name123"
    products = ["Ubuntu..", "OtherProduct"]
  }

patch_source {
    ... # Another patch source
  }
  ...
}
```

These will then translate to a list of patch_sources. Validation to be performed on:

patch_source
- type array of PatchSource objects
- Max of 20 defined

PatchSource.Configuration
- type string
- Max length of 1024

PatchSource.Name
- type string
- regex: `^[a-zA-Z0-9_\-.]{3,50}$`

PatchSource.Products
- type array of strings
- minimum of 1 item, maximum of 20
- string length between 1 and 128 chars

**Comment by bflad** [comment]
This functionality has been merged and will release with version 3.28.0 of the Terraform AWS Provider, later today. Thank you to @jdheyburn for the implementation. 👍
